content/blog/ggplot2-3-4-0-size-to-linewidth/index.Rmd

```diff
@@ -0,0 +1,310 @@
+title: Make your ggplot2 extension package understand the new linewidth aesthetic
+date: 2022-08-24
+author: Thomas Lin Pedersen
+description: >
+    The next release of ggplot2 will contain a number of internal improvements 
+    and fixes long-time inconsistencies. One of these are the conflation of 
+    point size and linewidth into the same aesthetic. This post will go into 
+    detail with how you can make your extension package work well with the new
+    linewidth aesthetic.
+photo:
+  url: https://unsplash.com/photos/GsZLXA4JPcM
+  author: Ricardo Gomez Angel
+# one of: "deep-dive", "learn", "package", "programming", "roundup", or "other"
+categories: [deep-dive] 
+tags: [ggplot2]
+editor_options: 
+  markdown: 
+    wrap: 72
+---
+
+```{r setup, include = FALSE}
+set.seed(340)
+options(rlib_message_verbosity = "verbose")
+options(lifecycle_verbosity = "warning")
+library(rlang)
+```
+
+We are hard at work finishing the next release of ggplot2. While this
+release is mostly about internal changes, there are a few quite user
+visible changes as well. One of these upends the idea that the `size`
+aesthetic is responsible for *both* the sizing of point/text and the
+width of lines. With the next release we will have a `linewidth`
+aesthetic to take care of the latter, while `size` will continue
+handling the former.
+
+There are many excellent reasons for this change, all of which will have
+to wait until the release post to be discussed. This blog post is for
+those that maintain an extension package for ggplot2 and are left
+wondering how they should respond to this --- if that is you, please
+read on!
+
+## The way it works
+
+Before going into technicalities we'll describe how it is intended to
+work. We are well aware that we can't just make a change that would
+instantly break everyone's code. So, we have gone to great length to
+make old code work as before while gently coercing users into adopting
+the news paradigm. For example, take a look at this piece of old code:
+
+```{r}
+library(ggplot2)
+
+ggplot(airquality) + 
+  geom_line(
+    aes(x = Day, y = Temp, size = Wind, group = Month), 
+    lineend = "round"
+  )
+```
+
+As you can see, ggplot2 detects the use of the `size` aesthetic and
+informs the user about the new `linewidth` aesthetic but otherwise
+proceeds as before, producing the expected plot. As expected,
+`scale_size()` also works in this situation:
+
+```{r}
+ggplot(airquality) + 
+  geom_line(
+    aes(x = Day, y = Temp, size = Wind, group = Month), 
+    lineend = "round"
+  ) + 
+  scale_size("Windspeed (mph)", range = c(0.5, 3))
+```
+
+but ultimately we of course wants users to migrate to the following
+code:
+
+```{r}
+ggplot(airquality) + 
+  geom_line(
+    aes(x = Day, y = Temp, linewidth = Wind, group = Month), 
+    lineend = "round"
+  ) + 
+  scale_linewidth("Windspeed (mph)", range = c(0.5, 3))
+```
+
+> The last two plots are not equal because the `default` `linewidth`
```

Comment:
Note that there's an important difference between these two plots (and one of the reasons we're making the change):